source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -1194,6 +1183,39 @@ def isochrone(self, request, instance):
 
         return resp
 
+    def handle_ridesharing_services(self, logger, instance, request, pb_response):
+
+        if instance.ridesharing_services and (
+            'ridesharing' in request['origin_mode'] or 'ridesharing' in request['destination_mode']
+        ):
+
+            def active_asynchronous_ridesharing(request, instance):
+                if request.get("_asynchronous_ridesharing", None) is None:
+                    # when the param is not set in the request, use the instance config
+                    return instance.asynchronous_ridesharing
+                else:
+                    return request.get("_asynchronous_ridesharing")
+
+            if not active_asynchronous_ridesharing(request, instance) or request.get('only_ridesharing', False):
+                logger.debug('trying to add ridesharing journeys')
+                try:
+                    decorate_journeys_with_ridesharing_offers(pb_response, instance, request)
+                except Exception:
+                    logger.exception('Error while retrieving ridesharing ads')
+                if request.get('only_ridesharing', False):
+                    for j in pb_response.journeys:
```

Comment:
if `_ridesharing_only` is on, we should remove all other kraken_calls which don't contain ridesharing... so that we can slightly reduce the load of the platform.

https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/scenarios/new_default.py#L126

Reply:
I let the all multimodal possiblities happen (without filtering or disabled kraken call).
Because i don't want to biaise the choice algorithm.

Reply:
you're right, we can do it later if we observe an obvious load problem